packages/slider/src/useSlider.ts

```diff
@@ -0,0 +1,317 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import { composeEventHandlers, KEYS } from '@zendeskgarden/container-utilities';
+import { IUseSliderProps, IUseSliderReturnValue } from './types';
+
+export const SLIDER_MIN = 0;
+export const SLIDER_MAX = 100;
+export const SLIDER_STEP = 1;
+
+export function useSlider<T extends Element = Element, M extends HTMLElement = HTMLElement>({
+  trackRef,
+  minThumbRef,
+  maxThumbRef,
+  min = SLIDER_MIN,
+  max = SLIDER_MAX,
+  defaultMinValue,
+  defaultMaxValue,
+  minValue,
+  maxValue,
+  onChange = () => undefined,
+  step = SLIDER_STEP,
+  jump = step,
+  disabled,
+  rtl,
+  environment
+}: IUseSliderProps<T, M>): IUseSliderReturnValue {
+  const doc = environment || document;
```

Comment:
With `environment` being `undefined`, this would throw a reference error in a non-browser environment. The same would apply for line 73 with `window`. This can be a possible followup item as we think through this problem.
![Screen Shot 2022-10-07 at 3 07 55 PM](https://user-images.githubusercontent.com/12474067/194636853-d93d4de4-3fc1-4cae-b071-add7d564cc66.png)

Reply:
Yeah, this is why `environment` needs to be defined for SSR usage. We can consider more elaborate handling, but I think it is fair to say that SSR needs to give Garden something (bare-bones) to go on. I agree we can consider as a possible follow-up if/when SSR hits this container.